HungarianAlgorithm/Application/ProgramOptions.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace Application
+{
+    public class ProgramOptions
+    {
+        public bool GenerateFile { get; private set; } = false;
+        public string Filename { get; private set; } = String.Empty;
+        public int N { get; private set; } = 2;
+        public int K { get; private set; } = 2;
```

Comment:
Czemu settery są prywatne, skoro i tak implementujemy je jako metody?

Reply:
Idea była taka, żeby zaznaczyć, że te pola są raczej readonly i nie możesz ich ustawić przypisując normalnie wartość do pola, a jedynie korzystając z metody. Najlepiej by było ustawić je w konstruktorze, ale czytamy z wejścia token za tokenem i trzeba gdzieś odkładać te wartości. Trochę też teraz czuję, że to jest na siłę przekombinowane

Reply:
Idea taka zła nie jest - można nawet zostawić ten konstruktor tak jak jest - po prostu settery też mogą mieć swoje body, w których ukryjemy tę logikę 🙂

Reply:
Ale wtedy też trzeba pisać getter :confused:

Reply:
No to prawda.. i backing field implementować... 😐